assembl/static2/js/app/components/administration/discussion/phaseForm.jsx

```diff
@@ -0,0 +1,147 @@
+
+  const splitButtonTitle = identifier ?
+    I18n.t(`administration.modules.${identifier}`) :
+    I18n.t('administration.timelineAdmin.singleModule');
+
+  return (
+    <div className="phase-form">
+      <div className="date-picker-field">
+        <div className="date-picker-type">
+          <Translate value="search.datefilter.from" />
+        </div>
+        <label htmlFor="start-datepicker" className="datepicker-label">
+          <DatePicker
+            placeholderText={startDatePickerPlaceholder}
+            selected={start}
+            id="start-datepicker"
+            onChange={onStartDateChange}
+            showTimeSelect
+            timeFormat="HH:mm"
+            dateFormat="LLL"
+            locale={locale}
+            shouldCloseOnSelect
+          />
+          <div className="icon-schedule-container">
+            <span className="assembl-icon-schedule grey" />
+          </div>
+        </label>
+      </div>
+      <div className="date-picker-field">
+        <div className="date-picker-type">
+          <Translate value="search.datefilter.to" />
+        </div>
+        <label htmlFor="end-datepicker" className="datepicker-label">
+          <DatePicker
+            placeholderText={endDatePickerPlaceholder}
+            id="end-datepicker"
+            selected={end}
+            onChange={onEndDateChange}
+            showTimeSelect
+            timeFormat="HH:mm"
+            dateFormat="LLL"
+            locale={locale}
+            shouldCloseOnSelect
+          />
+          <div className="icon-schedule-container">
+            <span className="assembl-icon-schedule grey" />
+          </div>
+        </label>
+      </div>
+      <div className="inline text-xs margin-l">
+        <Translate value="administration.timelineAdmin.phaseModule" />
+      </div>
+      <div className="margin-m">
+        <SplitButton
+          className="admin-dropdown"
+          id={`dropdown-${phaseId}`}
+          title={splitButtonTitle}
+          onSelect={handleIdentifierChange}
+        >
+          {modulesTranslationKeys.map(key => (
+            <MenuItem key={`module-${key}`} eventKey={key}>
+              {I18n.t(`administration.modules.${key}`)}
+            </MenuItem>
+          ))}
+        </SplitButton>
+      </div>
+      <div className="text-xs configure-module-text">
+        <Translate value="administration.timelineAdmin.configureModule" />
+        <Link to="" /* TODO: add route to phase configuration page */ className="configure-module-link">
+          <Translate value="administration.timelineAdmin.configureModuleLink" count={phaseNumber} />
+        </Link>
+      </div>
+    </div>
+  );
+};
+
+const mapStateToProps = (state, { phaseId }) => {
+  const phase = state.admin.timeline.phasesById.get(phaseId);
+  return {
+    identifier: phase ? phase.get('identifier') : null,
+    start: phase ? phase.get('start') : null,
```

Comment:
`moment` type can be null?

Reply:
Do you mean in terms of type ? I will add the possibility of being null in the declaration.
But here I did this because there is a placeHolder in the datepicker that shows when the value received is null